fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerConflictsTest.java

```diff
@@ -201,5 +201,60 @@ public void run() {
         assertSame(recordStore.getIndexState(index.getName()), IndexState.READABLE);
         assertEquals(0, timer.getCount(FDBStoreTimer.Events.COMMIT_FAILURE));
     }
+
+    @Test
+    public void testOnlineIndexerIdempotentNested() throws Exception {
+        // If a conflict happens, this test should succeed by retries
+        Random rand = new Random();
+        final long seed = rand.nextInt();
+        LOGGER.info("Random seed is " + seed);
+        rand = new Random(seed); // make rand repeatable (note that multi threading behaviour is not fully repeatable in this test)
+
+        List<TestRecords1Proto.MySimpleRecord> records =
+                LongStream.range(0, 400).mapToObj(val -> TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(val).setNumValue2((int)val + 1).build()
+                ).collect(Collectors.toList());
+
+        Index index = new Index("newIndex", field("num_value_2"), IndexTypes.VALUE);
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = metaDataBuilder -> metaDataBuilder.addIndex("MySimpleRecord", index);
+        openSimpleMetaData();
+        try (FDBRecordContext context = openContext()) {
+            for (int i = 50; i <= 150; i++) {
+                // even numbers 100-300 (that's 101 values)
+                recordStore.saveRecord(records.get(i * 2));
+            }
+            context.commit();
+        }
+
+        openSimpleMetaData(hook);
+        try (FDBRecordContext context = openContext()) {
+            recordStore.markIndexWriteOnly(index.getName()).join();
```

Comment:
Hm, it's weird that you need to mark the index as write only.... It _should_ be marking the index as write only when the record store is opened, I think, unless I'm misunderstanding how this all works.

Reply:
Omit the writeOnly but keeping the commit works!